pkg/util/mathutil/math.go

```diff
@@ -90,43 +87,38 @@ func Min[T constraints.Ordered](x T, xs ...T) T {
 	return minv
 }
 
-// Clamp restrict a value to a certain interval.
+// Clamp restricts a value to a certain range [minv, maxv].
 func Clamp[T constraints.Ordered](n, minv, maxv T) T {
-	if n >= maxv {
+	if n > maxv {
 		return maxv
-	} else if n <= minv {
+	}
+	if n < minv {
 		return minv
 	}
 	return n
 }
 
-// NextPowerOfTwo returns the smallest power of two greater than or equal to `i`
-// Caller should guarantee that i > 0 and the return value is not overflow.
+// NextPowerOfTwo returns the smallest power of two greater than or equal to `i`.
 func NextPowerOfTwo(i int64) int64 {
```

Comment:
I'm not sure the performance of these two versions. If you want to change it, please post some benchmark tests